src/factors/VelPose2D.jl

```diff
@@ -16,27 +16,50 @@ VelPose2VelPose2(z1::T1, z2::T2) where {T1 <: IIF.SamplableBelief, T2 <: IIF.Sam
 
 getSample(cf::CalcFactor{<:VelPose2VelPose2}, N::Int=1) = ([rand(cf.factor.Zpose.z,N);rand(cf.factor.Zvel,N)], )
 
-function (cf::CalcFactor{<:VelPose2VelPose2})(res::AbstractVector{<:Real},
-                                              meas,
+function IIF.getParametricMeasurement(s::VelPose2VelPose2{<:MvNormal, <:MvNormal}) 
+
+  meas = [mean(s.Zpose.z); mean(s.Zvel)]
+
+  iΣp = invcov(s.Zpose.z)
+  iΣv = invcov(s.Zvel)
+
+  iΣ = zeros(eltype(iΣp), 5,5)
+
+  iΣ[1:3,1:3] .= iΣp
+  iΣ[4:5,4:5] .= iΣv
+
+  return meas, iΣ
+end
+
+function (cf::CalcFactor{<:VelPose2VelPose2})(meas,
                                               Xi,
                                               Xj  )
   #
+  #FIXME JT - Createing new res for simplicity, it may not hold up well though
+  res = Vector{eltype(Xi)}(undef, 5)
```

Comment:
ah cool okay, let's try a few ways and that will hopefully help show how to do it natively in cfo.residual for all factors, available if users want to use it to reduce allocations.  Think it's okay to experiment with this for a bit